File: frontend/src/services/projects.ts
import api from './api'
import { Project, CreateProjectRequest } from '../types'

export const projectService = {
  async createProject(projectData: CreateProjectRequest): Promise<Project> {
    const response = await api.post('/projects', projectData)
    return response.data
  },

  async getProjects(): Promise<Project[]> {
    const response = await api.get('/projects')
    return response.data
  },

  async getProject(id: string): Promise<Project> {
    const response = await api.get(`/projects/${id}`)
    return response.data
  },

  async deleteProject(id: string): Promise<void> {
    await api.delete(`/projects/${id}`)
  },
}

File: backend/src/api/projects.rs
use actix_web::{web, HttpResponse, Result, Scope};

pub fn configure() -> Scope {
    web::scope("/projects")
        .route("", web::post().to(create_project))
        .route("", web::get().to(list_projects))
        .route("/{id}", web::get().to(get_project))
        .route("/{id}", web::delete().to(delete_project))
}

async fn create_project() -> Result<HttpResponse> {
    Ok(HttpResponse::Ok().json(serde_json::json!({
        "message": "Create project endpoint - database not connected yet"
    })))
}

async fn list_projects() -> Result<HttpResponse> {
    Ok(HttpResponse::Ok().json(serde_json::json!([])))
}

async fn get_project() -> Result<HttpResponse> {
    Ok(HttpResponse::Ok().json(serde_json::json!({
        "message": "Get project endpoint - database not connected yet"
    })))
}

async fn delete_project() -> Result<HttpResponse> {
    Ok(HttpResponse::Ok().json(serde_json::json!({
        "message": "Delete project endpoint - database not connected yet"
    })))
}